contracts/finance/VestingWallet.sol

```diff
@@ -154,10 +175,19 @@ contract VestingWallet is Context {
     function _vestingSchedule(uint256 totalAllocation, uint64 timestamp) internal view virtual returns (uint256) {
         if (timestamp < start()) {
             return 0;
-        } else if (timestamp > end()) {
+        } else if (timestamp >= end()) {
             return totalAllocation;
         } else {
             return (totalAllocation * (timestamp - start())) / duration();
         }
     }
+
+    /**
+     * @dev Reverts if the sender is not the {releaser} address.
+     */
+    function _checkReleaser() internal view virtual {
+        if (_msgSender() != releaser()) {
+            revert VestingWalletUnauthorizedReleaser(_msgSender());
+        }
```

Comment:
That is a major behavior change that I don't remember we agreed on. 

IMO, if we want that (I'm not sold) we need stronger documentation about that change. We also need to provide an easy path to re-implementing the current behavior (that doesn't involve overriding `_checkReleaser` without a super call).

Maybe something like "if releaser is 0, than anyone can release"
```solidity

```

Reply:
Before anyone proposes that, I really not a fan of advertizing

```solidity
    function releaser() public view virtual override returns (address) {
        return _msgSender();
    }
```

Reply:
(we need optionals values)

Reply:
I removed the commit with the `onlyReleaser` since we agreed on discussing further with the research team, let's see what alternative we can agree on :D